packages/components/src/tab-panel/test/index.tsx

```diff
@@ -34,7 +34,23 @@ const TABS = [
 
 const getSelectedTab = () => screen.getByRole( 'tab', { selected: true } );
 
+const originalGetClientRects = window.HTMLElement.prototype.getClientRects;
+
 describe( 'TabPanel', () => {
+	beforeAll( () => {
+		// Mocking `getClientRects()` is necessary to pass a check performed by
+		// the `focus.tabbable.find()` and by the `focus.focusable.find()` functions
+		// from the `@wordpress/dom` package.
+		// @ts-expect-error We're not trying to comply to the DOM spec, only mocking
+		window.HTMLElement.prototype.getClientRects = function () {
```

Comment:
Curious why we need this.. I saw it's also the case for some other tests like `NavigableMenu`, but I'd love to understand better. Is it related with the testing library used?

Reply:
`NavigableMenu` internally uses `focus.tabbable.find()` and `focus.focusable.find()` to search for the tabbable/focusable items — the logic requires a tabbable item [to be visible](https://github.com/WordPress/gutenberg/blob/feat/tab-panel-manual-tab-activation/packages/dom/build-module/focusable.js) — unfortunately, by default `js-dom` doesn't satisfy any of those checks, meaning that by default in our tests `NavigableMenu` won't move the selection to the next/previous tabs when interacting with the keyboard.

Mocking `getClientRects()` makes the above `isVisible` check pass and allows the test to work :)

Reply:
So it's related to `js-dom`.. Thanks for elaborating Marco!

Reply:
No worries! It took me a while to find out at first too 😅